things/service.go

```diff
@@ -928,18 +921,13 @@ func (ts *thingsService) UnassignThing(ctx context.Context, token string, groupI
 		return errors.ErrAuthorization
 	}
 
-	var chIDs []string
 	for _, thingID := range thingIDs {
 		ch, err := ts.channels.RetrieveByThing(ctx, user.GetId(), thingID)
 		if err != nil {
 			return err
 		}
 
-		chIDs = append(chIDs, ch.ID)
-	}
-
-	for _, chID := range chIDs {
-		if err := ts.channels.Disconnect(ctx, user.GetId(), chID, thingIDs); err != nil {
+		if err := ts.channels.Disconnect(ctx, user.GetId(), ch.ID, thingIDs); err != nil {
```

Comment:
You are Disconnecting `thingsIDs` from `ch.ID` but are you sure that all of them are connected to the same channel?